Automattic/pocket-casts-ios
# Re-add: Use content type to set extension for downloaded files

Re-add [fix to use content type to set file extension for audio files](https://github.com/Automattic/pocket-casts-ios/pull/1577)

I needed to re-add the columns on episode tables for this fix to work correctly.

I also updated the code use an existing MetadataTask class to update the contentType instead of doing it on the end of the upload.

In order to make sure this change didn't affect performance in queries because of Episode table changes I run some performance measurement using the Filters tab and measuring the method: `episodeCount(forFilter: EpisodeFilter, episodeUuidToAdd: String?)`

Here are the results I got [before and after these changes](https://docs.google.com/spreadsheets/d/1n16l_q1j-DKrk_kBxJIrbJFUYX1YRHv3vhHq8lV60kU/edit?usp=sharing) on a iPhone Xs

| Before | After |
| - | - |
|  0.3064338418  |   0.3009749147 | 

## To test

## Test if the fix worked

- Start the app on the simulator
- Search for the following podcast: `ondacer` on the discovery section
- Subscribe to the podcast
- Open the podcast and download one of the episodes
- Tap play
- The episode should play correctly
- Now play an episode without downloading it first and see if it works correctly
- Smoke test other podcasts, to see if this change doesn't brake anything on that front

## Test DB migrations because of fields `contentType` being added again

- Install a 7.62 build then migrate to this version 7.66
- Install a 7.61 build then migrate to this version 7.66
- Install a 7.64 build then migrate to this version 7.66

## Test the performance of the app because of database changes

- Setup the app to use a large database by cherry picking the last 4 commits in this branch: `hang/copy_database`
- Use the large user database with 500mb
- Use a real device
- Start the app
- Go to filters view
- Scroll around

## Checklist
